src/main/java/io/nats/client/support/JsonValue.java

```diff
@@ -0,0 +1,259 @@
+        this(null, null, null, null, l, null, null, null, null, null);
+    }
+
+    public JsonValue(double d) {
+        this(null, null, null, null, null, d, null, null, null, null);
+    }
+
+    public JsonValue(float f) {
+        this(null, null, null, null, null, null, f, null, null, null);
+    }
+
+    public JsonValue(BigDecimal bd) {
+        this(null, null, null, null, null, null, null, bd, null, null);
+    }
+
+    public JsonValue(BigInteger bi) {
+        this(null, null, null, null, null, null, null, null, bi, null);
+    }
+
+    public JsonValue(List<JsonValue> list) {
+        this(null, null, null, null, null, null, null, null, null, list);
+    }
+
+    public JsonValue(JsonValue... values) {
+        this(null, null, null, null, null, null, null, null, null,
+            values == null ? null : Arrays.asList(values));
+    }
+
+    private JsonValue(Map<String, JsonValue> map, String string, Boolean bool, Integer i, Long l, Double d, Float f, BigDecimal bd, BigInteger bi, List<JsonValue> array) {
+        this.map = map;
+        this.array = array;
+        this.string = string;
+        this.bool = bool;
+        this.i = i;
+        this.l = l;
+        this.d = d;
+        this.f = f;
+        this.bd = bd;
+        this.bi = bi;
+        if (i != null) {
+            this.type = Type.INTEGER;
+            number = i;
+            object = number;
+        }
+        else if (l != null) {
+            this.type = Type.LONG;
+            number = l;
+            object = number;
+        }
+        else if (d != null) {
+            this.type = Type.DOUBLE;
+            number = this.d;
+            object = number;
+        }
+        else if (f != null) {
+            this.type = Type.FLOAT;
+            number = this.f;
+            object = number;
+        }
+        else if (bd != null) {
+            this.type = Type.BIG_DECIMAL;
+            number = this.bd;
+            object = number;
+        }
+        else if (bi != null) {
+            this.type = Type.BIG_INTEGER;
+            number = this.bi;
+            object = number;
+        }
+        else {
+            number = null;
+            if (map != null) {
+                this.type = Type.OBJECT;
+                object = map;
+            }
+            else if (string != null) {
+                this.type = Type.STRING;
+                object = string;
+            }
+            else if (bool != null) {
+                this.type = Type.BOOL;
+                object = bool;
+            }
+            else if (array != null) {
+                this.type = Type.ARRAY;
+                object = array;
+            }
+            else {
+                this.type = Type.NULL;
+                object = null;
+            }
+        }
+    }
+
+    public String toString(Class<?> c) {
+        return toString(c.getSimpleName());
+    }
+
+    public String toString(String key) {
+        return "\"" + key + "\":" + toJson();
```

Comment:
Could optionally use `QUOTE + key + QUOTE + ":" + toJson()` here, like how `valueString` also uses it. But not a biggy.

Reply:
fixed